src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -192,11 +248,11 @@ public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connec
                     {
                         // If the URI is currently connecting from a different request, cancel it before we try to connect
                         CancellationTokenSource currentSource;
```

Comment:
Looks like I was wrong in my above comment - current behavior is to cancel current connection if we get a new one in? I guess we can preserve this logic...